### PR TITLE
Migrate to Actions v2

### PR DESCRIPTION
**Pull request by smockle**
Migrates from GitHub Actions v1 to Actions v2.

**Rationale:**
- The [documentation](https://help.github.com/en/articles/getting-started-with-github-actions) now presupposes Actions v2 usage
- The “Workflows defined in **.github/workflows**” link in the “Actions” tab is currently broken (screenshot below):
workflows-link

**Changes:**
- Move workflow file into the `.github/workflows` directory
- Replace [HCL with YAML](https://help.github.com/en/articles/migrating-github-actions-from-hcl-syntax-to-yaml-syntax)
- Remove unneeded `uses = "actions/npm@v2.0.0"` (see the [Node.js sample workflow](https://github.com/actions/starter-workflows/blob/master/ci/node.js.yml))

### CODE REVIEW

**Comment by smockle** [comment]
To eliminate future merge conflicts, I’d originally attempted to incorporate the changes in #450 into this PR. But I was stumped by 401s during attempts to publish to GPR, e.g. https://github.com/primer/components/commit/409e586758051b62243b677ed7d715ccabf52236/checks. I spent some time debugging, then decided to keep this PR small and focused on the migration to Actions v2.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+      - name: Publish to the npm registry
+        uses: "primer/publish@v1.0.0"
+        with:
+          args: "-- --unsafe-perm --allow-same-version"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
+
+      - name: Deploy
```
Comment (emplums) [inline .github/workflows/main.yml]: Can you actually remove this deploy action? We use the Now integration now :)

**Comment by emplums** [comment]
I'm not super worried about keeping the publish to GPR stuff - we can circle back on that later and re-set it up! It didn't take too long :)

File: .github/workflows/main.yml (hunk 1 above)
Comment (smockle) [reply .github/workflows/main.yml]: Now it’s gone! Fixed in https://github.com/primer/components/pull/543/commits/ecc4a53f5cd9cb9e8da85bfe497fcf525408ee6d

**Pull request merged**